lms/djangoapps/completion/api/v1/views.py

```diff
@@ -135,3 +142,28 @@ def post(self, request, *args, **kwargs):
             }, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
         return Response({"detail": _("ok")}, status=status.HTTP_200_OK)
+
+
+class CompletionReportView(APIView):
```

Comment:
According to how open-edx handles the reports generation, the request to generate a report should be a post request, and the payload of this request should contain the required information to create the report (in this case, the course_id). The task status can be consulted via get request with no problems. I think we can use separate views to process reports generations and task status

Reply:
I agree with the first part but I don't understand the point of separate the views when  open-edx uses the same view to handle multiple request

Reply:
we have 2 endpoints: one for triggering the reports generation, the other one to consult the status of a celery task. I think they are different enough to consider using separated views, the purposes are different

Reply:
yes, but also it's  the status of the report generation and when it is done returns  the file or json .... For me it' is like  TeamsListView that allows to get or create a course team ---> get or create report  the only different this is async...

Reply:
OK, this discussion can be extended for so long, take my approach or yours (I think you will take yours), but make it clear and readable